ddcommon/src/unix_utils.rs

```diff
@@ -188,17 +188,21 @@ fn is_being_traced() -> io::Result<bool> {
     // Check to see whether we are being traced.  This will fail on systems where procfs is
     // unavailable, but presumably in those systems `ptrace()` is also unavailable.
     // The caller is free to treat a failure as a false.
-    let file = File::open("/proc/self/status")?;
-    let reader = BufReader::new(file);
-
-    for line in reader.lines() {
-        let line = line?;
-        if line.starts_with("TracerPid:") {
-            let tracer_pid = line.split_whitespace().nth(1).unwrap_or("0");
-            return Ok(tracer_pid != "0");
+    // This function may run in signal handler, so we should ensure that we do not allocate
+    // memory on the heap (ex: avoiding using BufReader for exao).
+    let mut file = File::open("/proc/self/status")?;
+    let mut buf = [0u8; 8192];
+    let n = file.read(&mut buf)?;
+    let data = &buf[..n];
+
+    for line in data.split(|&b| b == b'\n') {
+        if let Ok(line_str) = std::str::from_utf8(line) {
```

Comment:
IF this fails, should we return false, or an error?

Reply:
The question would be what should we do on error.
Today, the semantic is:
- `Ok(true)` : use `CLONE_PTRACE` as an extra-flags
- `Ok(false)`: use `0` to specify no clone flags

Should an error have the same behavior as `Ok(false)` or should be stop and no crash report is generated ?
All that said, I would say, we are fine today :)

Reply:
so keep it like that